receiver/googlecloudmonitoringreceiver/receiver.go

```diff
@@ -5,51 +5,203 @@ package googlecloudmonitoringreceiver // import "github.com/open-telemetry/opent
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/multierr"
 	"go.uber.org/zap"
+
+	"google.golang.org/api/iterator"
+	"google.golang.org/genproto/googleapis/api/metric"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudmonitoringreceiver/internal"
+
+	"cloud.google.com/go/monitoring/apiv3/v2/monitoringpb"
+	
+	monitoring "cloud.google.com/go/monitoring/apiv3/v2"
 )
 
 type monitoringReceiver struct {
-	config *Config
-	logger *zap.Logger
-	cancel context.CancelFunc
+	config         *Config
+	logger         *zap.Logger
+	client         *monitoring.MetricClient
+	metricsBuilder *internal.MetricsBuilder
+	startOnce      sync.Once
 }
 
 func newGoogleCloudMonitoringReceiver(cfg *Config, logger *zap.Logger) *monitoringReceiver {
 	return &monitoringReceiver{
-		config: cfg,
-		logger: logger,
+		config:         cfg,
+		logger:         logger,
+		metricsBuilder: internal.NewMetricsBuilder(logger),
 	}
 }
 
-func (m *monitoringReceiver) Scrape(ctx context.Context) (pmetric.Metrics, error) {
-	// Dummy use to fix lint errors
-	ctx.Deadline()
-	metrics := pmetric.NewMetrics()
-	m.logger.Debug("Scrape metrics ")
+func (mr *monitoringReceiver) Start(ctx context.Context, _ component.Host) error {
+	var startErr error
+	mr.startOnce.Do(func() {
+		client, err := monitoring.NewMetricClient(ctx)
+		if err != nil {
+			startErr = fmt.Errorf("failed to create a monitoring client: %w", err)
+			return
+		}
+
+		mr.client = client
+	})
 
-	return metrics, nil
+	return startErr
 }
 
-func (m *monitoringReceiver) Start(ctx context.Context, _ component.Host) error {
-	ctx, m.cancel = context.WithCancel(ctx)
-	err := m.initialize(ctx)
-	if err != nil {
-		return err
-	}
+func (mr *monitoringReceiver) Shutdown(context.Context) error {
+	mr.logger.Debug("shutting down googlecloudmonitoringreceiver receiver")
 	return nil
 }
 
-func (m *monitoringReceiver) Shutdown(context.Context) error {
-	m.logger.Debug("shutting down googlecloudmonitoringreceiver receiver")
-	return nil
+func (mr *monitoringReceiver) Scrape(ctx context.Context) (pmetric.Metrics, error) {
+	var (
+		calStartTime         time.Time
+		calEndTime           time.Time
+		filterQuery          string
+		allTimeSeriesMetrics []*monitoringpb.TimeSeries
+		gErr                 error
+	)
+
+	// Iterate over each metric in the configuration to calculate start/end times and construct the filter query.
+	for _, metric := range mr.config.MetricsList {
+		// Define the interval and delay times
+		interval := mr.config.CollectionInterval
+		delay := metric.Delay
+
+		// Calculate the start and end times
+		calStartTime, calEndTime = calculateStartEndTime(interval, delay)
+
+		// Get the filter query for the metric
+		filterQuery = getFilterQuery(metric)
+
+		// Log an error if the filter query is empty
+		if filterQuery == "" {
+			mr.logger.Error("Internal Server Error")
```

Comment:
we should probably `continue` here so that we don't make the query

Reply:
Remove the condition, as validation on the config variable will directly trigger an error from the test cases.